Sources/Purchasing/Configuration.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Configuration.swift
+//
+//  Created by Joshua Liebowitz on 5/6/22.
+
+import Foundation
+
+/**
+ * ``Configuration`` can be used when configuring the ``Purchases`` instance. It is not required to be used, but
+ * highly recommended. This class follows a builder pattern.
+ *
+ * To configure your `Purchases` instance using this object, follow these steps.
+ *
+ * **Steps:**
+ * 1. Call ``Configuration/configurationBuilder(withAPIKey:)`` To obtain a ``ConfigurationBuilder`` object.
+ * 2. Set this builder's properties using the "`with(`" functions.
+ * 3. Call ``ConfigurationBuilder/build()`` to obtain the `Configuration` object.
+ * 4. Pass the `Configuration` object into ``Purchases/configure(withConfiguration:)``.
+ *
+ */
+@objc(RCConfiguration) public class Configuration: NSObject {
+
+    static let storeKitTimeoutSecondsDefault: Int = 30
+    static let storeKitRequestTimeoutDefault = DispatchTimeInterval.seconds(storeKitTimeoutSecondsDefault)
+    static let networkTimeoutSecondsDefault: Int = 60
+
+    let apiKey: String
+    let appUserID: String?
+    let observerMode: Bool
+    let userDefaults: UserDefaults?
+    let storeKit2Setting: StoreKit2Setting
+    let dangerousSettings: DangerousSettings?
+    let networkTimeoutSeconds: Int
+    let storeKit1TimeoutSeconds: Int
+
+    override public init() {
+        fatalError("Use static function configurationBuilder(withAPIKey:) to configure.")
+    }
+
+    private init(withConfigurationBuilder configurationBuilder: ConfigurationBuilder) {
+        self.apiKey = configurationBuilder.apiKey
+        self.appUserID = configurationBuilder.appUserID
+        self.observerMode = configurationBuilder.observerMode
+        self.userDefaults = configurationBuilder.userDefaults
+        self.storeKit2Setting = configurationBuilder.storeKit2Setting
+        self.dangerousSettings = configurationBuilder.dangerousSettings
+
+        if configurationBuilder.storeKit1TimeoutSeconds < 5 {
+            self.storeKit1TimeoutSeconds = 5
+        } else {
+            self.storeKit1TimeoutSeconds = configurationBuilder.storeKit1TimeoutSeconds
+        }
+
+        if configurationBuilder.networkTimeoutSeconds < 5 {
+            self.networkTimeoutSeconds = 5
+        } else {
+            self.networkTimeoutSeconds = configurationBuilder.networkTimeoutSeconds
+        }
+    }
+
+    /// Factory method for the ``ConfigurationBuilder`` object that is required to create a `Configuration`
+    @objc public static func configurationBuilder(withAPIKey apiKey: String) -> ConfigurationBuilder {
```

Comment:
```suggestion
    @objc public static func builder(withAPIKey apiKey: String) -> Builder {
```

So this becomes `let builder: Configuration.Builder = Configuration.builder(withAPIKey:)` instead of `let builder: Configuration.ConfigurationBuilder = Configuration.configurationBuilder(withAPIKey:)`

Reply:
✅